test/unit/components/bn-as-decimal-input-test.js

```diff
@@ -48,4 +48,40 @@ describe('BnInput', function () {
       checkValidity () { return true } },
     })
   })
+  
+  it('can tolerate wei precision', function (done) {
+    const renderer = ReactTestUtils.createRenderer()
+
+    let valueStr = '1000000000000000000'
+
+    const value = new BN(valueStr, 10)
+
+    const inputStr = '1000000000.000000001'
+
+    let targetStr = '1000000000000000001'
+
+    const target = new BN(targetStr, 10)
+
+    const precision = 9 // ether precision
```

Comment:
Slight misnomer for the test, ether precision is 18, but the test should still work at other scales.